### PR TITLE
[native pos] Simplify replicate-nulls-and-any logic

### PR DESCRIPTION
**Pull request by mbasmanova**
When replicate-nulls-and-any is true, PartitionAndSerialize operator emits an
extract boolean column which tells ShuffleWrite operator whether to replicate a
row to all partitions.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp
Comment (vermapratyush) [inline presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp:1]: On line 30, can you update the comment.
```
/// The output of this operator has 3 columns:
/// (1) partition number (INTEGER);
/// (2) serialized row (VARBINARY)
/// (3) is row replicated (if replicateNullsAndAny_==true) (BOOL)
```

**Review by vermapratyush — APPROVED** [review]
Thanks for the PR. This looks much cleaner.

**Review by ajaygeorge — APPROVED** [review]

File: presto-native-execution/presto_cpp/main/operators/LocalPersistentShuffle.h [hunk 1]
```diff
@@ -105,20 +104,20 @@ class LocalPersistentShuffleWriter : public ShuffleWriter {
       const std::string& root,
       int32_t partition) const;
 
+  // Used to make sure files created by this thread have unique names.
+  const std::thread::id threadId_;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/operators/LocalPersistentShuffle.h:108]: Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova thanks for change!

**Pull request merged**